dask/tests/test_optimization.py

```diff
@@ -227,6 +227,48 @@ def test_fuse_keys():
     )
 
 
+def test_donot_substitute_same_key_multiple_times():
```

Comment:
```suggestion
def test_donot_substitute_same_key_multiple_times() -> None:
```
If you add `-> None` to new tests mypy will be able to check them as well. I think it's good practice to just do it if you are worrying about the type hints. 
It helps catch the edge cases that aren't always obvious when just type hinting the functions.